lib/manageiq/appliance_console/database_admin.rb

```diff
@@ -153,6 +168,36 @@ def ask_custom_file_options(server_uri)
         @task_params = ["--", params]
       end
 
+      def ask_swift_file_options
+        require 'uri'
+        @uri              = ask_for_uri(*remote_file_prompt_args_for("swift"))
+        user              = just_ask(USER_PROMPT)
+        pass              = ask_for_password("password for #{user}")
+        region            = just_ask("OpenStack Swift Region")
+        port              = just_ask("OpenStack Swift Port", "5000")
+        security_protocol = ask_with_menu(*security_protocol_menu_args)
+        api_version       = ask_with_menu(*api_version_menu_args)
+        domain_ident      = just_ask("OpenStack V3 Domain Identifier") if api_version == "v3"
+
+        @task          = "evm:db:#{action}:remote"
+        @uri           = URI.parse("#{URI(@uri).scheme}://#{URI(@uri).host}:#{port}#{URI(@uri).path}")
```

Comment:
What is happening here? I'm not sure what this line is trying to accomplish, but my gut feeling is that it can be done in a better way.

Reply:
To add to this, this is basically parsing the `@uri` 4 times, and it seems pretty unecessary.  I think doing the following would suffice:

```ruby
@uri              = URI(ask_for_uri(*remote_file_prompt_args_for("swift")))
# ...
@uri.port         = just_ask("OpenStack Swift Port", "5000")
```

And then you can remove this line we are commenting on entirely.

Reply:
Adding the port into the URI.  I'm sure there are *other* ways to do it.  I am open to suggestions other than this one.